File: packages/android/connector/src/main/java/app/quiltt/connector/QuilttConnectorEvent.kt
package app.quiltt.connector

data class ConnectorSDKCallbackMetadata(
    val connectorId: String,
    val profileId: String?,
    val connectionId: String?
)

typealias ConnectorSDKOnEventCallback = (ConnectorSDKEventType, ConnectorSDKCallbackMetadata) -> Unit
typealias ConnectorSDKOnEventExitCallback = (ConnectorSDKEventType, ConnectorSDKCallbackMetadata) -> Unit

typealias ConnectorSDKOnExitSuccessCallback = (ConnectorSDKCallbackMetadata) -> Unit
typealias ConnectorSDKOnExitAbortCallback = (ConnectorSDKCallbackMetadata) -> Unit
typealias ConnectorSDKOnExitErrorCallback = (ConnectorSDKCallbackMetadata) -> Unit

enum class ConnectorSDKEventType(val value: String) {
    Load("loaded"),
    ExitSuccess("exited.successful"),
    ExitAbort("exited.aborted"),
    ExitError("exited.errored")
}

File: packages/ios/Sources/QuilttConnector/QuilttConnectorEvent.swift
import Foundation

public struct ConnectorSDKCallbackMetadata {
    public let connectorId: String
    public let profileId: String?
    public let connectionId: String?
}

public typealias ConnectorSDKOnEventCallback = (ConnectorSDKEventType, ConnectorSDKCallbackMetadata)
    -> Void
public typealias ConnectorSDKOnEventExitCallback = (
    ConnectorSDKEventType, ConnectorSDKCallbackMetadata
) -> Void

public typealias ConnectorSDKOnExitSuccessCallback = (ConnectorSDKCallbackMetadata) -> Void
public typealias ConnectorSDKOnExitAbortCallback = (ConnectorSDKCallbackMetadata) -> Void
public typealias ConnectorSDKOnExitErrorCallback = (ConnectorSDKCallbackMetadata) -> Void

public enum ConnectorSDKEventType: String {
    case Load = "loaded"
    case ExitSuccess = "exited.successful"
    case ExitAbort = "exited.aborted"
    case ExitError = "exited.errored"
}
